build_tools/scripts/run_markdownlint.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Runs Markdownlint on Markdown (.md) files
+# https://github.com/igorshubovych/markdownlint-cli
+
+set -xuo pipefail
+
+declare -a included_files_patterns=(
+  # All .md files (disabled while we decide how rigorously to apply lint checks)
+  # "./**/*.md"
+
+  # Just .md files for the user-facing website.
+  "./docs/website/**/*.md"
+)
+
+declare -a excluded_files_patterns=(
+  "/third_party/"
+  "**/node_modules/**"
+)
+
+markdownlint "${included_files_patterns[*]}" \
+    --config ./docs/.markdownlint.yml \
+    --ignore "${excluded_files_patterns[*]}"
```

Comment:
@GMNGeoffrey other lint scripts run on only changed files... want that here also?

https://github.com/iree-org/iree/blob/2211d97ed36d98522313b36d28c0bb28e3c800ca/build_tools/scripts/run_buildifier.sh#L29-L44

(I tried to copy that logic over but my local runs kept saying no files changed... maybe I'd need to tweak my git index or base ref to test?)

Reply:
Uh that should work, although looking at it, it should be useing the merge base instead. I need to fix all our lint stuff... It would be nice to limit to only changed files, yes, but I think it's not the end of the world.